### ISSUE TITLE
Type.Extends is not compatible with Typescript?

### ISSUE THREAD

**Issue by ExpandOcean** [sinclairzx81/typebox#649]
Hi, it appears that when comparing two object schemas, the behavior for optional properties is different with Typescript.
Typebox version: 0.31.20

``` javascript
const schemaA = Type.Object({
  a: Type.Number(),
  b: Type.Number(),
});

const schemaB = Type.Object({
  b: Type.Number(),
  c: Type.Optional(Type.Number()),
});

const schemaAExtendsB = Type.Extends(
  schemaA,
  schemaB,
  Type.Literal(true),
  Type.Literal(false)
);

// SchemaAExtendsB is true
type SchemaAExtendsB = Static<typeof schemaAExtendsB>;

// schemaAExtendsB is
// {
//     "const": false,
//     "type": "boolean"
// }
console.log(schemaAExtendsB);
```

**Comment by sinclairzx81** [comment]
@ExpandOcean Hi

> it appears that when comparing two object schemas, the behavior for optional properties is different with Typescript.

This is a good catch. Have published a fix on 0.31.21. 

Note that the Type.Extends type is "mostly" TypeScript aligned (as good as I've been able to get it so far), but there may be a few edge cases here and there (which I'm documenting for future revisions). This optional case was definitely a bug, so fixing on the current revision makes sense. 

Let me know if this resolves the issue your side.
Cheers
S

**Comment by sinclairzx81** [comment]
@ExpandOcean All good, will close up this issue for now :)

Thanks for reporting.
